lxd/operations.go

```diff
@@ -257,14 +257,44 @@ func operationDelete(d *Daemon, r *http.Request) response.Response {
 			projectName = api.ProjectDefaultName
 		}
 
+		// Separate resources by entity type. If there are multiple entries of a particular entity type we can reduce
```

Comment:
Can we standardise this pattern into a helper function  we can call in each appropriate place?

Reply:
This is similar to `project.FilterUsedBy` except that this one returns an error if any are unauthorized - rather than filtering them out.

I suppose here I could write a method `FilterByEntitlement`, then `FilterUsedBy` would call it with `can_view` and this function would call it and check if the length of the input list is equal to the length of the output list. I think that might be less clear though. Not sure :thinking: